src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExporter.cs

```diff
@@ -62,75 +63,115 @@ internal DynatraceMetricsExporter(DynatraceExporterOptions options, ILogger<Dyna
 				enrichWithDynatraceMetadata: _options.EnrichWithDynatraceMetadata);
 		}
 
-		public override async Task<ExportResult> ExportAsync(IEnumerable<Metric> metrics, CancellationToken cancellationToken)
+		public override ExportResult Export(in Batch<Metric> batch)
 		{
-			// split all metrics into batches of DynatraceMetricApiConstants.PayloadLinesLimit lines
-			var chunked = metrics
-				.Select((val, i) => new { val, batch = i / DynatraceMetricApiConstants.PayloadLinesLimit })
-				.GroupBy(x => x.batch)
-				.Select(x => x.Select(v => v.val));
-
-			var exportResults = new List<ExportResult>();
-
-			foreach (var chunk in chunked)
+			// Prevents the exporter's HTTP operations from being instrumented.
+			using (var scope = SuppressInstrumentationScope.Begin())
 			{
-				var httpRequest = new HttpRequestMessage(HttpMethod.Post, _options.Url);
-				var sb = new StringBuilder();
+				var metricLines = GetSerializeMetricLines(batch);
 
-				foreach (var metric in chunk)
-				{
-					SerializeMetric(sb, metric);
-				}
+				// split all metrics into batches of DynatraceMetricApiConstants.PayloadLinesLimit lines
+				var chunked = metricLines
+					.Select((val, i) => new { val, batch = i / DynatraceMetricApiConstants.PayloadLinesLimit })
+					.GroupBy(x => x.batch)
+					.Select(x => x.Select(v => v.val));
 
-				var metricLines = sb.ToString();
-				_logger.LogDebug(metricLines);
+				ExportResult exportResult;
 
-				// If there are no metric lines, we do not need to send a request.
-				if (metricLines.Length == 0)
+				foreach (var chunk in chunked)
 				{
-					return ExportResult.Success;
-				}
+					var httpRequest = new HttpRequestMessage(HttpMethod.Post, _options.Url);
+					var joinedMetricLines = string.Join(Environment.NewLine, chunk);
 
-				httpRequest.Content = new StringContent(metricLines);
-				try
-				{
-					var response = await _httpClient.SendAsync(httpRequest);
-					if (response.IsSuccessStatusCode)
+					if (joinedMetricLines.Length == 0)
 					{
-						_logger.LogDebug("StatusCode: {StatusCode}", response.StatusCode);
-						exportResults.Add(ExportResult.Success);
+						return ExportResult.Success;
 					}
-					else
+
+					httpRequest.Content = new StringContent(joinedMetricLines);
+					try
 					{
-						_logger.LogError("StatusCode: {StatusCode}", response.StatusCode);
-						_logger.LogError("Content: {Content}", await response.Content.ReadAsStringAsync());
-						exportResults.Add(ExportResult.FailedNotRetryable);
+						// sync over async isn't great, but the exporter is executed in it's own thread.
+						var response = _httpClient.SendAsync(httpRequest).GetAwaiter().GetResult();
```

Comment:
```suggestion
						var response = _httpClient.SendAsync(httpRequest).ConfigureAwait(false).GetAwaiter().GetResult();
```
There is no reason the synchronization context needs to be attached here is there?  I think chaining `ConfigureAwait(false)` to the invocation of this task may increase the performance of this call.

Reply:
The `Export` method here is not async, so `GetWaiter().GetResult()` is really just blocking the thread. More on it here: https://devblogs.microsoft.com/dotnet/configureawait-faq/#im-using-getawaiter-getresult-do-i-need-to-use-configureawaitfalse.

If it were async, then `ConfigureAwait(false)` would make sense :)

Reply:
That blog confirms my question: 
_It doesn’t mean, however, that there will never be a custom SynchronizationContext or TaskScheduler present. If some user code (or other library code your app is using) sets a custom context and calls your code, or invokes your code in a Task scheduled to a custom TaskScheduler, then even in ASP.NET Core your awaits may see a non-default context or scheduler that would lead you to want to use ConfigureAwait(false)._

Also from that same post: 
_This leads to the general guidance of: if you’re writing general-purpose library code, use ConfigureAwait(false)._ 

It may not block the main thread but it is being called in a foreach loop. This could cause the child thread that is executing this method to take exponentially longer to finish.

Reply:
Yes, that is correct. It is the "norm" to always use `ConfigureAwait(false)` in library code even when you know there's "no context" (e.g. in ASP.NET Core apps) essentially because it could run in other places where it would make a difference. 

But in this specific case, the method is not `async`, so there's no callback to be called as the thread will be blocked and it's intentional. From the blog post above:

> I’m using **GetAwaiter().GetResult()**. Do I need to use ConfigureAwait(false)?
>
>**No**. ConfigureAwait only affects the callbacks. Specifically, the awaiter pattern requires awaiters to expose an IsCompleted property, a GetResult method, and an OnCompleted method (optionally with an UnsafeOnCompleted method). ConfigureAwait only affects the behavior of {Unsafe}OnCompleted, so if you’re just directly calling to the awaiter’s GetResult() method, whether you’re doing it on the TaskAwaiter or the ConfiguredTaskAwaitable.ConfiguredTaskAwaiter makes zero behavior difference. So, if you **see task.ConfigureAwait(false).GetAwaiter().GetResult()** in code, you can replace it with **task.GetAwaiter().GetResult()** (and also consider whether you really want to be blocking like that).


Making the async http request run synchronously (blocking essentially) here is intended. The `Export` method here is always called by a MetricReader in a separated thread (https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs#L53), so the code here in the foreach will block that thread until it completes. From the OpenTelemetry [Metrics SDK spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#exportbatch):

> `Expor`t will never be called concurrently for the same exporter instance. Export can be called again only after the current call returns.

Also, the same behavior (no `ConfigureAwait(false)`) can be seen in other exporters in the OpenTelemetry SDK: [ZipkinExporter](https://github.com/open-telemetry/opentelemetry-dotnet/blob/6e6afdbc3614cfc52d20825a25f06bd80b5ba64e/src/OpenTelemetry.Exporter.Zipkin/ZipkinExporter.cs), [OtlpHttpExporter](https://github.com/open-telemetry/opentelemetry-dotnet/blob/2a221722a1c46b47d0f6e860a7a96597d496628f/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/BaseOtlpHttpExportClient.cs#L80)

Reply:
Simply put, there is no need to capture the synchronization context here so `ConfigureAwait(false)` makes sense to implement. It has nothing to do with sync/async. The code will work fine the way it is but you may be able to grab some inches here.

Reply:
We have ideas to add some benchmarks, specially there's probably a lot of optimizations that can be done around strings since we create a lot of them. I'll keep this in mind and if we have a baseline of benchmarks we can measure and see if there's any improvement when applying `ConfiguredAwait(false)`. Thanks for taking a look!